src/atomate2/classical_md/openmm/jobs/base.py

```diff
@@ -0,0 +1,494 @@
+"""The base class for OpenMM simulation makers."""
+
+from __future__ import annotations
+
+import copy
+import time
+import warnings
+from dataclasses import dataclass, field
+from datetime import datetime
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, NoReturn
+
+from emmet.core.classical_md.openmm import (
+    Calculation,
+    CalculationInput,
+    CalculationOutput,
+    OpenMMTaskDocument,
+)
+from jobflow import Maker, Response
+from mdareporter.mdareporter import MDAReporter
+from openff.interchange import Interchange
+from openmm import Integrator, LangevinMiddleIntegrator, Platform
+from openmm.app import StateDataReporter
+from openmm.unit import kelvin, picoseconds
+
+from atomate2.classical_md.core import openff_job
+from atomate2.classical_md.utils import increment_name, task_reports
+
+if TYPE_CHECKING:
+    from openmm.app.simulation import Simulation
+
+OPENMM_MAKER_DEFAULTS = {
```

Comment:
What is the benefit of having these here rather than settings them in the `BaseOpenMMMaker` directly? It seems like this would be cleaner?

Reply:
I agonized over this but it was ultimately the only solution I could think of. I'll respond here but the answer would be the same for this comment too:

> I think this would be even simpler if the defaults are on the base class, as then there are only two places to check.

When users are defining workflows I wanted to support easy inheritance between jobs. Priority is: settings from current maker > settings from previous task input >  a default value. If I set the defaults in BaseOpenMMMaker directly, there is no way of differentiating between settings from the current maker and the defaults. Maybe there is another solution, but the only one I could think of was external defaults. 

You can see the inheritance logic below.

```python
        if getattr(self, attr, None) is not None:
            attr_value = getattr(self, attr)  # if the defaults were set directly this would always return them.
        elif getattr(prev_input, attr, None) is not None:
            attr_value = getattr(prev_input, attr)
        else:
            attr_value = defaults.get(attr)
```